src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1984,7 +1984,7 @@ public virtual bool Accepts(DType type, out KeyValuePair<string, DType> schemaDi
                 case DKind.OptionSetValue:
                     accepts = (type.Kind == Kind &&
                                 (OptionSetInfo == null || type.OptionSetInfo == null || type.OptionSetInfo == OptionSetInfo)) ||
-                               type.Kind == DKind.Unknown;
+                               type.Kind == DKind.Unknown || (type.IsMultiSelectOptionSet() && (OptionSetInfo != null && type.TypeTree.GetPairs() != null && type.TypeTree.GetPairs().First().Value?.OptionSetInfo == OptionSetInfo));
```

Comment:
>type.TypeTree.GetPairs().First().Value? [](http://example.com/codeflow?start=157&length=39)

Same here. This can't be null right if type.IsMultiSelectOptionSet()